source/plugin/data/custom/serialization.rst

```diff
@@ -2,4 +2,118 @@
 Serializing Custom Data
 =======================
 
-.. TODO Stuff about DataManipulatorBuilders for DataManipulators, and DataBuilders for serializables
\ No newline at end of file
+.. javadoc-import::
+    org.spongepowered.api.data.DataHolder
+    org.spongepowered.api.data.DataQuery
+    org.spongepowered.api.data.manipulator.DataManipulator
+    org.spongepowered.api.data.persistence.AbstractDataBuilder
+    org.spongepowered.api.data.persistence.DataBuilder
+    org.spongepowered.api.data.persistence.DataContentUpdater
+    org.spongepowered.api.data.persistence.DataManipulator
+    org.spongepowered.api.data.persistence.DataTranslator
+    org.spongepowered.api.data.serialization.DataSerializable
+
+Without a method serializing and deserializng, your data will not persist across restarts. Sponge has a few different ways to serialize/deserialize data based on the type of data:
+
+- :javadoc:`DataSerializable`\ s implement an interface to perform serialization, and use :javadoc:`DataBuilder` for deserialization and creation
+- :doc:`DataManipulators <datamanipulators>` also implement ``DataSerializable``, but instead use a :javadoc:`DataManipulator` for deserialization and creation
+- Other objects, particularly those that do not or cannot implement ``DataSerializable``, use :javadoc:`DataTranslator` for both serialization and deserialization
+
+This means that practically any object in Java can be saved to disk if it has registered way to be deserialized!
+
+Reading DataViews
+=================
+
+Whenever you're reading a serialized object, it's tempting to read all the individual values yourself in order to create the objects for your data. However, depending on the data saved in the container there's a few ways ways that are far more convenient:
```

Comment:
Wow, this is a long line. The guidelines suggest something shorter...
Also please change ``there's`` to ``there are`` (it's a plural).